jigsaw/client/menu.py

```diff
@@ -0,0 +1,167 @@
+            rectangle = Rect((right_edge - 50, count), button_size)
+            button = pygame_gui.elements.UIButton(
+                relative_rect=rectangle, text=mode.title(), manager=manager)
+            count += 83
+            setattr(self, f"{mode}_button", button)
+
+        self.right_panel = pygame_gui.elements.UIPanel(
+            relative_rect=Rect((WIDTH - 200, 0), (200, HEIGHT)),
+            starting_height=0,
+            manager=manager
+        )
+
+    def draw_completion_bar(self, completion_percent, position, size):
+        """ Draws a completion bar on the screen.
+        Args:
+            screen (pygame.Surface): The surface to draw on.
+            completion_percent (float): The completion percentage (0.0.to 1.0)
+            position (tuple): (x,y) position ef the top-left corner of the bar.
+            size (tuple): (width, height) of the bar.
+        """
+        border_color = (0, 0, 0)
+        fill_color = (0, 255, 0)
+
+        draw.rect(screen, border_color, position + size, 2)
+
+        fill_width = int(size[0] * completion_percent)
+        draw.rect(screen, fill_color, (position[0], position[1], fill_width, size[1]))
+
+    def event_handler(self, e):
+        if e.type == QUIT:
+            quit()
+        if e.type == USEREVENT:
+            if e.user_type == pygame_gui.UI_BUTTON_PRESSED:
+                if e.ui_element == self.quick_button:
+                    self.active_view = self.view_instances["play_game"]
+                if e.ui_element == self.secret_button:
+                    self.active_view = self.view_instances["play_game"]
+                if e.ui_element == self.settings_button:
+                    self.active_view = self.view_instances["settings"]
+                    self.active_view.show()
+                if e.ui_element == self.credits_button:
+                    self.active_view = self.view_instances["credits"]
+                    self.active_view.show()
+        return True
+
+    def mainloop(self):
+        while True:
+            screen.fill(WHITE)
+            time_delta = self.clock.tick(60) / 1000.0
+            for e in event.get():
+                if not self.active_view.event_handler(e):
+                    self.active_view.hide()
+                    self.active_view = self
+                manager.process_events(e)
+            manager.draw_ui(screen)
+            manager.update(time_delta)
+            display.update()
+            display.flip()
+
+
+def hamming_distance(current_positions, correct_positions):
+    """Calculate the Hamming distance between the current puzzle positions and the correct positions."""
+    return sum(current != correct for current, correct in zip(current_positions, correct_positions))
+
+
+def simulate_tile_positions(total_tiles):
+    """
+    Simulate random tile position to produce random Hamming distance
+    """
+    return random.sample(range(total_tiles), total_tiles)
+
+
+# Views
+
+def play_game(difficulty):
```

Comment:
```suggestion
def play_game(difficulty) -> dict[str, str]:
```
Please add what the type of `difficulty` is. I can't see where this function is used currently so not sure on the type.